### PR TITLE
Add rlwrap

### PR DESCRIPTION
**Pull request by r7st**
Adding rlwrap package. 

From README:
"rlwrap is a 'readline wrapper', a small utility that uses the GNU Readline library to allow the editing of keyboard input for any command."

https://github.com/hanslub42/rlwrap

### CODE REVIEW

**Review by oetiker — APPROVED** [review]

File: build/rlwrap/build.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# Copyright 2022 r7st r7st.guru@gmail.com
+
+. ../../lib/build.sh
+
+PROG=rlwrap
+VER=0.45.2
+PKG=ooce/util/rlwrap
+SUMMARY="readline wrapper"
+DESC="rlwrap is a 'readline wrapper', a small utility that uses the GNU Readline library to allow the editing of keyboard input for any command."
```
Comment (citrus-it) [inline build/rlwrap/build.sh:23]: Could you please wrap this to 80 columns? A lot of packages do something like this:
```suggestion
DESC="rlwrap is a 'readline wrapper', a small utility that uses the GNU "
DESC+="Readline library to allow the editing of keyboard input for any command."
```

File: build/rlwrap/build.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+set_arch 64
+
+set_mirror "$GITHUB/hanslub42/$PROG/releases/download"
+set_checksum sha256 "9f8870deb46e473d21b5db89d709b6497f4ef9fa06d44eebc5f821daa00c8eca"
```
Comment (citrus-it) [inline build/rlwrap/build.sh:28]: I've put the sources on the omnios mirror now, along with a checksum file. Please can you remove these two lines?

File: build/rlwrap/build.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+CFLAGS+=" -fstack-protector -gdwarf"
```
Comment (citrus-it) [inline build/rlwrap/build.sh:30]: You shouldn't need to set these explicitly, the framework takes care of adding the appropriate flags for stack protector and DWARF that is compatible with the CTF format, and varies depending on which omnios release is being built for. I just tried this on r151042 and see:

```
/opt/gcc-11/bin/gcc -DHAVE_CONFIG_H -I. -I..    -DDATADIR=\"/opt/ooce/share\" 
    -O2 -fno-aggressive-loop-optimizations
    -fstack-protector-strong -fstack-protector
    -gdwarf -gdwarf-2 -gstrict-dwarf
    -m64 -MT ptytty.o -MD -MP -MF .deps/ptytty.Tpo -c -o ptytty.o ptytty.c
```
so the framework is already adding `-fstack-protector-strong` and `-gdwarf-2 -gstrict-dwarf` and then converting the DWARF to the native CTF data. I think this line should just be deleted.

File: build/rlwrap/build.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+init
+download_source v$VER $PROG-$VER
+prep_build
+build
+strip_install
```
Comment (citrus-it) [inline build/rlwrap/build.sh:36]: Please remove this line (`strip-install`). The framework takes care of converting the DWARF to CTF and then removing it from the binary, so it ends up effectively stripped.

File: build/rlwrap/build.sh [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+set_arch 64
+
+set_mirror "$GITHUB/hanslub42/$PROG/releases/download"
+set_checksum sha256 "9f8870deb46e473d21b5db89d709b6497f4ef9fa06d44eebc5f821daa00c8eca"
+
+CFLAGS+=" -fstack-protector -gdwarf"
+
+init
+download_source v$VER $PROG-$VER
```
Comment (citrus-it) [inline build/rlwrap/build.sh:33]: With the sources on the omnios mirror, this should change to:
```suggestion
download_source $PROG $PROG $VER
```

File: build/rlwrap/local.mog [hunk 6]
```diff
@@ -0,0 +1,21 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+# CDDL HEADER START
```
Comment (citrus-it) [inline build/rlwrap/local.mog:10]: This is a stray line
```suggestion
```

**Review by citrus-it — CHANGES_REQUESTED** [review]
Thanks for the PR, and for contributing to OmniOS.

I just have a few changes I'd like to see before it gets merged, please seem the comments in-line.

File: build/rlwrap/build.sh (hunk 2 above)
Comment (r7st) [reply build/rlwrap/build.sh:28]: Thanks, I'm able to pull from the mirror.

**Comment by r7st** [comment]
@citrus-it Revised as requested. Please let me know if anything else needs to be modified!

**Review by citrus-it — APPROVED** [review]

**Pull request merged**